Error when trying to deploy from subdirectory

I don't know if it's a real bug but @KannarFr told me to report it.

To replicate :

Create folder with git init
then create a subfolder and link clever app
When I run clever deploy I get a error message

`[ERROR] Cannot read property 'split' of null`
 
I think it's just a missing control when .git folder is missing or on parent directory

--- a/src/command-promise-handler.js
+++ b/src/command-promise-handler.js
@@ -1,6 +1,5 @@
 'use strict';
 
-const _ = require('lodash');
 const Logger = require('./logger.js');
 
 function handleCommandPromise (promise) {

--- a/src/logger.js
+++ b/src/logger.js
@@ -36,7 +36,7 @@ const Logger = _(['debug', 'info', 'warn', 'error'])
     const consoleFn = (severity === 'error') ? console.error : console.log;
     const { prefix, prefixLength } = getPrefix(severity);
     return [severity, (err) => {
-      const message = _.get(err, 'message', err)
+      const message = _.get(err, 'message', err);
       const formattedMsg = formatLines(prefixLength, processApiError(message));
       if (process.env['CLEVER_VERBOSE'] && severity === 'error') {
         console.error('[STACKTRACE]');

--- a/src/models/app_configuration.js
+++ b/src/models/app_configuration.js
@@ -10,6 +10,7 @@ const slugify = require('slugify');
 const { conf } = require('./configuration.js');
 const Logger = require('../logger.js');
 
+// TODO: Maybe use fs-utils findPath()
 function loadApplicationConf (ignoreParentConfig = false, pathToFolder) {
   if (pathToFolder == null) {
     pathToFolder = path.dirname(conf.APP_CONFIGURATION_FILE);

--- a/src/models/fs-utils.js
+++ b/src/models/fs-utils.js
@@ -1,0 +1,22 @@
+'use strict';
+
+const { join } = require('path');
+const fs = require('fs');
+const { promisify } = require('util');
+
+const fsStat = promisify(fs.stat);
+
+function findPath (dir, name) {
+  const fullPath = join(dir, name);
+  return fsStat(fullPath)
+    .then(() => dir)
+    .catch((e) => {
+      if (e.code === 'ENOENT' && dir !== '/') {
+        const parent = join(dir, '..');
+        return findPath(parent, name);
+      }
+      throw e;
+    });
+}
+
+module.exports = { findPath };

--- a/src/models/git.js
+++ b/src/models/git.js
@@ -7,12 +7,17 @@ const Bacon = require('baconjs');
 const git = require('isomorphic-git');
 const { autocomplete } = require('cliparse');
 
-const { loadOAuthConf } = require('./configuration.js');
 const slugify = require('slugify');
+const { findPath } = require('./fs-utils.js');
+const { loadOAuthConf } = require('./configuration.js');
 
-const repo = { fs, dir: '.' };
+async function getRepo () {
+  const dir = await findPath('.', '.git');
+  return { fs, dir };
+}
 
 async function addRemote (remoteName, url) {
+  const repo = await getRepo();
   const safeRemoteName = slugify(remoteName);
   const allRemotes = await git.listRemotes({ ...repo });
   const existingRemote = _.find(allRemotes, { remote: safeRemoteName });
@@ -25,7 +30,9 @@ function resolveFullCommitId (commitId) {
   if (commitId == null) {
     return Bacon.constant(null);
   }
-  return Bacon.fromPromise(git.expandOid({ ...repo, oid: commitId }))
+  const fullCommitIdPromise = getRepo()
+    .then((repo) => git.expandOid({ ...repo, oid: commitId }));
+  return Bacon.fromPromise(fullCommitIdPromise)
     .flatMapError((e) => {
       if (e.code === 'ShortOidNotFound') {
         return new Bacon.Error(`Commit id ${commitId} is ambiguous`);
@@ -35,6 +42,7 @@ function resolveFullCommitId (commitId) {
 }
 
 async function getRemoteCommit (remoteUrl) {
+  const repo = await getRepo();
   const tokens = await loadOAuthConf().toPromise();
   const remoteInfos = await git.getRemoteInfo({
     ...repo,
@@ -45,18 +53,21 @@ async function getRemoteCommit (remoteUrl) {
   return _.get(remoteInfos, 'refs.heads.master');
 }
 
-function getFullBranch (branchName) {
+async function getFullBranch (branchName) {
+  const repo = await getRepo();
   if (branchName === '') {
     return git.currentBranch({ ...repo, fullname: true });
   }
   return git.expandRef({ ...repo, ref: branchName });
 };
 
-function getBranchCommit (refspec) {
+async function getBranchCommit (refspec) {
+  const repo = await getRepo();
   return git.resolveRef({ ...repo, ref: refspec });
 }
 
 async function push (remoteUrl, branchRefspec, force) {
+  const repo = await getRepo();
   const tokens = await loadOAuthConf().toPromise();
   try {
     const push = await git.push({
@@ -82,7 +93,8 @@ async function push (remoteUrl, branchRefspec, force) {
 }
 
 function completeBranches () {
-  return git.listBranches(repo)
+  return getRepo()
+    .then((repo) => git.listBranches(repo))
     .then(autocomplete.words);
 }
 

